test/onnx/onnx_test_common.py

```diff
@@ -424,11 +426,22 @@ def _compare_pytorch_onnx_with_ort(
     )
     ort_outputs = run_ort(onnx_program, onnx_format_args)
 
-    if len(ref_outputs) != len(ort_outputs):
+    # When model is a torch.export.ExportedProgram, the number of outputs in the ONNX model can be greater
```

Comment:
Create an output adapter step for this?

Reply:
It doesn't work, at least how we designed ` InputAdapter`  and `OutputAdapter`. They were designed to be used *after* the model is exported, to adapt input/output from pytorch to match the ONNX input/output.

The output adapter expand the pytorch output to match the ONNX output schema, so we could at most match the number of outputs. Thus, an extra output would be prepended to pytorch output to match the extra output on onnx graph.

However, the ONNX output will have an output with the *mutated* buffer while pytorch will have a not-mutated buffer as output, causing a mismatch between them.

Now if we decide to hack the purpose of the `OutputAdapter` and use it before the FX-ONNX to truncate the extra outputs, we could make it work. This is a break of contract, though.

Makes sense?

Reply:
I am creating a `adapt_onnx_outputs_to_torch` in a separate PR to help bridging this gap. Despite the output truncation, we will also need to update the mutated buffers within the model so that buffers match during tests

Reply:
done